File: testbed_ruby/test/attributes/immutable_test.rb

require 'test/unit'
require './src-gen-umple/door_a'
require './src-gen-umple/door_b'
require './src-gen-umple/door_c'
require './src-gen-umple/door_g'

module CruiseAttributesTest
class ImmutableTest < Test::Unit::TestCase

  def test_Immutable
    door = DoorC.new("1",2,3.4,Date.parse("1978-12-05"),Time.parse("10:11:15"),false,DoorB.new(5))

    assert_equal("1",door.get_id)
    assert_equal false, door.respond_to?("set_id")
  
    assert_equal(2,door.get_intId)
    assert_equal false, door.respond_to?("set_intId")
  
    assert_equal(3.4,door.get_doubleId)
    assert_equal false, door.respond_to?("set_doubleId")

    assert_equal(Date.parse("1978-12-05"),door.get_dateId)
    assert_equal false, door.respond_to?("set_dateId")

    assert_equal(Time.parse("10:11:15"),door.get_timeId)
    assert_equal false, door.respond_to?("set_timeId")

    assert_equal(false,door.get_booleanId)
    assert_equal false, door.respond_to?("set_booleanId")
    
    assert_equal(5,door.get_doorId.get_id)
    assert_equal false, door.respond_to?("set_doorId")
  end
  
  def test_ImmutableInitialized
    door = DoorA.new
    assert_equal("1",door.get_id)
    assert_equal false, door.respond_to?("set_id")
  
    assert_equal(2,door.get_intId)
    assert_equal false, door.respond_to?("set_intId")
  
    assert_equal(3.4,door.get_doubleId)
    assert_equal false, door.respond_to?("set_doubleId")
    
    assert_equal(Date.parse("1978-12-05"),door.get_dateId)
    assert_equal false, door.respond_to?("set_dateId")

    assert_equal(Time.parse("10:11:15"),door.get_timeId)
    assert_equal false, door.respond_to?("set_timeId")

    assert_equal(false,door.get_booleanId)
    assert_equal false, door.respond_to?("set_booleanId")
    
    assert_equal(5,door.get_doorId.get_id)
    assert_equal false, door.respond_to?("set_doorId")
  end

  def test_LazyImmutable
    door = DoorG.new
    #assert_nil(door.get_doorId)
    assert_nil(door.get_dateId)
    assert_nil(door.get_timeId)

    doorId = DoorB.new(5)
    assert_equal(true,door.set_doorId(doorId)) 
    assert_equal(true,door.set_dateId(Date.parse("1978-12-05")))
    assert_equal(true,door.set_timeId(Time.parse("10:11:15")))

    assert_equal(door.get_doorId,doorId)
    assert_equal(door.get_dateId,Date.parse("1978-12-05"))
    assert_equal(door.get_timeId,Time.parse("10:11:15"))

    assert_equal(false,door.set_doorId(DoorB.new(5))) 
    assert_equal(false,door.set_dateId(Date.parse("1978-12-05")))
    assert_equal(false,door.set_timeId(Time.parse("10:11:15")))
  end

end
end

File: testbed_python/test/attributes/immutable_test.py
import unittest

# Test classes are imported dynamically
# Must be generated into ImportModules' namespace then imported
from ImportModules import importModules

importModules(["DoorA", "DoorB", "DoorC"], ["attributes", "test"])
from ImportModules import *
import datetime


class ImmutableTest(unittest.TestCase):
    @unittest.skip("Update with different Date/Time")
    def test_Immutable(self):
        door = DoorC.DoorC(
            "1",
            2,
            3.4,
            datetime.datetime.strptime("1234", "%Y").date(),
            datetime.datetime.strptime("1235", "%H%M"),
            False,
            DoorB.DoorB(5),
        )

        self.assertEqual("1", door.getId())

        self.assertEqual(2, door.getIntId())

        self.assertEqual(3.4, door.getDoubleId(), 0.01)

        self.assertEqual(
            datetime.datetime.strptime("1234", "%Y").date(),
            door.getDateId(),
            "DateTime object did not match",
        )

        self.assertEqual(
            datetime.datetime.strptime("1235", "%H%M"),
            door.getTimeId(),
            "DateTime object did not match",
        )

        self.assertEqual(False, door.getBooleanId())

        self.assertEqual(DoorB.DoorB(5), door.getDoorId())

    @unittest.skip("Update with different Date/Time")
    def test_ImmutableInitialized(self):
        door = DoorA.DoorA()
        self.assertEqual("1", door.getId())

        self.assertEqual(2, door.getIntId())

        self.assertEqual(3.4, door.getDoubleId(), 0.01)

        self.assertEqual(Date(1234), door.getDateId())

        self.assertEqual(Time(1235), door.getTimeId())

        self.assertEqual(False, door.getBooleanId())

        self.assertEqual(DoorB(5), door.getDoorId())
